internal/services/authorization/role_definition_data_source.go

```diff
@@ -152,11 +150,13 @@ func (a ArmRoleDefinitionDataSource) Read() sdk.ResourceFunc {
 				return err
 			}
 
-			if config.Name == "" && config.RoleDefinitionId == "" {
-				return fmt.Errorf("one of `name` or `role_definition_id` must be specified")
+			defId := config.RoleDefinitionId
+			scope := config.Scope
+
+			if scope == "" {
```

Comment:
Is this a fix for the old code? Or we shall enforce it in the schema?

Reply:
scope is an optional property, when it's not specified, we take the subscription id as scope.

Reply:
I mean in its initial implementation, it didn't give this default value. My question is that is there a valid case that the data source can work with an empty `scope`?

By checking the implementation of the `Get` and `List` of the old SDK, the empty `scope` seems has no special handling, in which case it results into tenant scope?

Reply:
yes, it's a valid case.

Reply:
Then does it mean defaulting the `scope` to the subscription id will then stop the `tenant` scope work?

Reply:
yes, updated to use the empty one.